src/metaswitch/clearwater/config_manager/plugin_base.py

```diff
@@ -65,3 +71,11 @@ def on_config_changed(self, value, alarm):
         should also report the status of the controlled file to the alarm
         manager."""
         pass
+
+    @abstractmethod
+    def on_creating_etcd_key(self, value):
```

Comment:
I don't think you need this method, I think it should all Just Work without it. Let's discuss on Wednesday.